gpt4all-chat/src/database.cpp

```diff
@@ -2524,6 +2534,35 @@ void Database::retrieveFromDB(const QList<QString> &collections, const QString &
             results->append(tempResults.value(id));
 }
 
+bool Database::ftsIntegrityCheck()
+{
+    bool isConsistent = false;
+
+    QSqlQuery q(m_db);
+    if (!q.prepare("insert into chunks_fts(chunks_fts, rank) values('integrity-check', 1);")) {
```

Comment:
We should use global constants for these query strings, as we already do with the others.

Also, wouldn't it be good to check for SQLITE_CORRUPT_VTAB specifically (possibly via nativeErrorCode)? This way we could also avoid the need for the separate prepare() because we can disambiguate between preparation failure and a corrupt table.